topic-operator/src/main/java/io/strimzi/operator/topic/model/Results.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.model;
+
+import io.strimzi.api.kafka.model.common.Condition;
+import io.strimzi.api.kafka.model.topic.ReplicasChangeStatus;
+import org.apache.kafka.clients.admin.AlterConfigOp;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+/**
+ * Reconciliation results.
+ */
+public class Results {
+    // results from internal operations
+    private final Map<ReconcilableTopic, Either<TopicOperatorException, Object>> results = new HashMap<>();
+    
+    // pending status updates
+    private final Map<ReconcilableTopic, Collection<Condition>> conditions = new HashMap<>();
+    private final Map<ReconcilableTopic, ReplicasChangeStatus> replicasChanges = new HashMap<>();
+
+    // pending alter config changes
+    private List<Pair<ReconcilableTopic, Collection<AlterConfigOp>>> configChanges = new ArrayList<>();
+    
+    /**
+     * @param partitionedByError Results partitioned by error.
+     */
+    public void setResults(PartitionedByError<ReconcilableTopic, ? extends Object> partitionedByError) {
+        setRightResults(partitionedByError.ok());
+        setLeftResults(partitionedByError.errors());
+    }
+
+    /**
+     * @param ok Success results.
+     */
+    public void setRightResults(Stream<? extends Pair<ReconcilableTopic, ? extends Object>> ok) {
+        ok.forEach(pair -> setResult(pair.getKey(), Either.ofRight(null)));
+    }
+
+    /**
+     * @param ok Success results.
+     */
+    public void setRightResults(Collection<ReconcilableTopic> ok) {
+        ok.forEach(rt -> setResult(rt, Either.ofRight(null)));
+    }
+
+    /**
+     * @param errors Error results.
+     */
+    public void setLeftResults(Stream<Pair<ReconcilableTopic, TopicOperatorException>> errors) {
+        errors.forEach(pair -> setResult(pair.getKey(), Either.ofLeft(pair.getValue())));
+    }
+
+    private void setResult(ReconcilableTopic key,
```

Comment:
I don't think set is the right word for this, because it doesn't definitely set the given result. `maybeSetResult()` might be better. Or something to suggest the non-trivial semantics, like `accrueResult` or something. 

This comment can also be applied to the other `set*()` methods, which should also be renamed.